<launch>
  <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="launch_prefix" default="" />

  <!-- configure camera input -->
  <!-- SET  camera_index for launch-->
  <arg name="camera_index" default="2" />

  <!--  -->
  <arg name="image_topic" default="image_raw" />
  <arg name="queue_size" default="1" />
  <arg name="camera_name" default="/camera$(arg camera_index)" />

  <!-- apriltag_ros continuous detection node -->
  <node pkg="apriltag_ros" type="gulliview_noetic_stream_cameraX_node"
    name="Stream_Camera1_detector"
    clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />

    <param name="publish_tag_detections_image" type="bool" value="false" />     <!-- default: false -->


    <param name="camera_index" type="int" value="$(arg camera_index)" />
    <param name="queue_size" type="int" value="$(arg queue_size)" />


    <param name="cam0_fx" type="double" value="1145.256303" />
    <param name="cam0_fy" type="double" value="1141.658316" />
    <param name="cam0_cx" type="double" value="961.614618" />
    <param name="cam0_cy" type="double" value="514.490771" />

    <param name="cam1_fx" type="double" value="1148.845354" />
    <param name="cam1_fy" type="double" value="1136.968276" />
    <param name="cam1_cx" type="double" value="987.148727" />
    <param name="cam1_cy" type="double" value="523.721356" />

    <param name="cam2_fx" type="double" value="1146.495853" />
    <param name="cam2_fy" type="double" value="1146.011083" />
    <param name="cam2_cx" type="double" value="953.976032" />
    <param name="cam2_cy" type="double" value="571.868765" />

    <param name="cam3_fx" type="double" value="1135.208474" />
    <param name="cam3_fy" type="double" value="1133.531325" />
    <param name="cam3_cx" type="double" value="998.932805" />
    <param name="cam3_cy" type="double" value="530.084624" />


    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find apriltag_ros)/config/object_setting.yaml" />
    <rosparam command="load" file="$(find apriltag_ros)/config/object.yaml" />
  </node>
</launch>